tests/drivers/spi/spi_slave_master_loopback/src/spi.c

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <spi.h>
+
+#define SPI_MASTER_CS (&spi_master_cs)
+#define SPI_SLAVE_CS (&spi_slave_cs)
+
+/* As configured in the board/.prj fragment */
+#define SPI_MASTER_DRV_NAME "SPI_LOOPBACK_MASTER"
+#define SPI_SLAVE_DRV_NAME "SPI_LOOPBACK_SLAVE"
+
+#if defined(CONFIG_BOARD_NUCLEO_F091RC)
+#define SPI_SLAVE 0
+#define MIN_FREQ 500000
```

Comment:
I guess you'd rather not make these application-specific Kconfig options added in with KCONFIG_ROOT, as discussed in the previous version -- any reason why not?

The advantage to doing it that way is to configure additional boards just with fragments, instead of having to touch this file.

Reply:
Ah, sorry, I think I see your point now:

> I did the DRV_NAME trick, I think the other stuff will need to wait until DT has been implemented for SPI

I'm personally OK with deferring this portability consideration until DT for SPI has been defined, especially if @tbursztyka has an idea for this. And even otherwise, I think it's better to have this code in tree than not.

So even though I like having a Kconfig solution that can be replaced with DT better than having a source-only solution, I'm OK with this change.

Reply:
OK, I added the Kconfig in the last push, please tell me if it's ok